tests/nightly/test_np_large_array.py

```diff
@@ -875,34 +875,65 @@ def test_digamma():
                 rtol=1e-3, atol=1e-5)
 
 @use_np
-@pytest.mark.skip(reason='broken on large tensors; also backward errors out')
-def test_rnn():
+def test_rnn_dim_check():
+    L_SEQ, BAT, L_INP, L_STA = 2**31, 4, 2**10, 2
+    data = np.random.uniform(-1, 1, (L_SEQ, BAT, L_INP))
```

Comment:
Why random input. Because then we are forced to use npx.waitall()

Reply:
If we initialized the data and params to 1 then the output will explode very fast; on the contrary if we initialize them to 0 then the output will be all zeros. In practice neither is an actual use case. I think testing testing rnn is tricky in that it's hard to establish a numerically correct reference.

Reply:
You can consider to revise it like what I have suggested. You generate 20 random numbers from uniform(-1, 1) and create a small array to store these numbers and cache it in the python script. Then, you cyclically use these numbers to construct the large tensor.

Reply:
> You can consider to revise it like what I have suggested. You generate 20 random numbers from uniform(-1, 1) and create a small array to store these numbers and cache it in the python script. Then, you cyclically use these numbers to construct the large tensor.

Right this would speed up testing, but we are still unable to test correctness without a ground truth. Out side the scope of this pr I definitely agree that more correctness tests can be added; there if we start with smaller tensors then we can just use a fixed seed to begin with. We can see the need for progressing to testing large tensors then